main.cpp

```diff
@@ -188,214 +286,313 @@ struct Range
         bool supportsWifi = true;
         bool isAnalog = false;
 
-        void informTime (bool timeUpdated);
-        void indicateRepairDate (int date, float needsRepair);
-        bool selfCleans (std::string brand, bool settingsOn);
+        std::string informCurrentTime (bool timeUpdated);
+        void indicateRepairDate (std::string date, bool needsRepair);
+        bool selfCleans (bool settingsOn = true);
         
     };
 
-    // 1) consumes fuel 
     void consumeFuel ( std::string fuelSource, bool rangeOn);
-    // 2) breaks down
     void breaksDown (int ageOfHeatingElement);
-    // 3) boil water
     int heatsTheKitchen (float durationOperated);
 
     RangeControls updatedSettings;
 };
-/*
-Thing 3) shopping cart
- */
+std::string Range::RangeControls::informCurrentTime (bool timeUpdated)
+{
+    std::string updatedTime, currentTime;
+    if(timeUpdated) return updatedTime;
+    return currentTime;
+
+}
+void Range::RangeControls::indicateRepairDate (std::string date, bool needsRepair)
+{
+    if(needsRepair)
+    {
+        std::cout << "Range needs repair by " << date << std::endl;
+    }
+    else
+    {
+    std::cout << "Range In Good Shape" << std::endl;
+    }
+    
+}
+bool Range::RangeControls::selfCleans (bool settingsOn)
+{
+    while(settingsOn) return settingsOn;
+    return !settingsOn;
+}
```

Comment:
loops have not been covered yet.
remove this `while` condition.
Also, think about the logic here.  
if settingsOn is true, you're going to return immediately.  
The loop will run at most once. 
Thus, there is no need for the loop at all.

delete everything inside this function.
write some std::cout statements to this function, based on what `settingsOn` is set to. 

- [ ] revise